palette/src/rgb/rgb.rs

```diff
@@ -953,14 +954,50 @@ where
     }
 }
 
+impl FromStr for Rgb {
+    type Err = Box<dyn std::error::Error>;
```

Comment:
This will have to change to something that isn't `Box`ed, due to the `#[no_std]` support requirement. A simple error type like

```rust
enum FromHexError {
    ParseIntError(ParseIntError),
    HexFormatError(&'static str), // Like "invalid hex code length" or "unexpected hex code prefix"
}
```

with all the impls it would need, would do the trick.

Reply:
I struggle with this. Sorry, I am fairly new to rust, but there is no Error type in core only in std.

Do you have a suggestion how to do this ?

Reply:
Ah, ok. No worries! :) You are right, it seems like the `Error` trait is only in `std`. In that case we will have to make use of conditional compilation if we want to use it (it's good practice to implement `Error`, so we should). `#[no_std]` is a particularly prickly feature. If we use `std` anywhere, unconditionally, it's all out the window, and it's hard to notice unless it's used on a platform that doesn't support the `std` library. I have set up a CI test to detect it, and that's what made this PR fail the tests.

For this case we have to give up the convenience of having the boxed trait in either case, since `Box` is also not available, so the first step would be to add an enum like the one I suggested and change this line like this:

```suggestion
    type Err = FromHexError;
```

That way, the parsing will work both with and without `std`.

Then the `FromHexError` type will need some traits (I would look at `ParseIntError` for inspiration), including `Error`. That's where we will have to check if the library user is allowing the `std` library:

```rust
#[cfg(feature = "std")] // Checks if the "std" feature from Cargo.toml is enabled
impl std::error::Error for FromHexError {
    // `std` is safe to use in this `impl` item, because of the `cfg` attribute.
    // It will not be included at all if the "std" feature is disabled.
    //
    // ...
}
```

The same goes for any other trait that is not in `core`.